shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1755,18 +1757,20 @@ bool WebContents::IsDevToolsFocused() {
 }
 
 void WebContents::EnableDeviceEmulation(
-    const blink::WebDeviceEmulationParams& params) {
+    const blink::DeviceEmulationParams& params) {
   if (type_ == Type::REMOTE)
     return;
 
   auto* frame_host = web_contents()->GetMainFrame();
   if (frame_host) {
-    auto* widget_host =
-        frame_host ? frame_host->GetView()->GetRenderWidgetHost() : nullptr;
-    if (!widget_host)
-      return;
-    widget_host->Send(new WidgetMsg_EnableDeviceEmulation(
-        widget_host->GetRoutingID(), params));
+    auto* widget_host_impl =
+        frame_host ? static_cast<content::RenderWidgetHostImpl*>(
```

Comment:
oof, i don't love casting to the `Impl`. Can we do this without reaching into the internals?

Reply:
Unfortunately, the emulation apis are only reachable via the `Impl` class https://source.chromium.org/chromium/chromium/src/+/master:content/browser/devtools/protocol/emulation_handler.cc;l=553-563

this should be safe to cast https://source.chromium.org/chromium/chromium/src/+/master:content/browser/renderer_host/render_widget_host_impl.cc;l=510-512